paddle/fluid/framework/framework.proto

```diff
@@ -91,52 +91,54 @@ message OpProto {
   required string comment = 5;
 }
 
-enum DataType {
-  BOOL = 0;
-  INT16 = 1;
-  INT32 = 2;
-  INT64 = 3;
-  FP16 = 4;
-  FP32 = 5;
-  FP64 = 6;
-}
+message VarType {
+  enum Type {
+    // POD types
+    BOOL = 0;
+    INT16 = 1;
+    INT32 = 2;
+    INT64 = 3;
+    FP16 = 4;
+    FP32 = 5;
+    FP64 = 6;
+
+    // other types, that might need additonal descriptions
+    LOD_TENSOR = 7;    // LoDTensor
+    SELECTED_ROWS = 8; // Tensor
+    FEED_MINIBATCH = 9;
+    FETCH_LIST = 10;
+    STEP_SCOPES = 11;
+    LOD_RANK_TABLE = 12;
+    // LOD_TENSOR_ARRAY = 7; // replaced by array-of-tensors. c.f. ArrayType
```

Comment:
LOD_TENSOR_ARRAY is a statically typed array. All tensors in the array will hold the same type and the same shape in compile time.

Reply:
Good point. So LoDTensorArray is not Arary<Tenosr>. We got that. Thanks for the reminder!